src/app/reporting/ReportScheduler.h

```diff
@@ -167,6 +167,18 @@ class ReportScheduler : public ReadHandler::Observer
 
     virtual void ReportTimerCallback() = 0;
 
+    void OnActiveModeEntered()
```

Comment:
Please document what this is for, and why it is needed (i.e. what happens if you don't call this: are there reports that will never be sent?)